clojure/src/rabbitmq/tutorials/rpc_client.clj

```diff
@@ -0,0 +1,42 @@
+(ns rabbitmq.tutorials.rpc-client
+  (:require [langohr.core :as lc]
+            [langohr.channel :as lch]
+            [langohr.queue :as lq]
+            [langohr.basic :as lb]
+            [langohr.consumers :as lcons])
+  (:import [com.rabbitmq.client QueueingConsumer]))
+
+(def ^{:const true} q "rpc_queue")
+
+(defn correlation-id-equals? [correlation-id d]
+  (= (.getCorrelationId (.getProperties d)) correlation-id))
+
+(defrecord FibonacciClient [conn ch cbq consumer]
+  clojure.lang.IFn
+  (invoke [this n]
+    (let [correlation-id (str (java.util.UUID/randomUUID))]
+      (lb/publish ch "" q (str n) {:reply-to cbq
+                                   :correlation-id correlation-id})
+      (lb/consume ch cbq consumer)
+      (-> (first (filter (partial correlation-id-equals? correlation-id)
+                         (lcons/deliveries-seq consumer)))
+          .getBody
+          (String. "UTF-8")
+          (read-string))))
+  java.io.Closeable
+  (close [this]
+    (.close conn)))
+
+(defn make-fibonacci-rpc-client []
+  (let [conn     (lc/connect)
+        ch       (lch/open conn)
+        cbq      (lq/declare ch "" {:auto-delete false :exclusive true})
+        consumer (QueueingConsumer. ch)]
```

Comment:
In part because tutorials code needs to be as straightforward as possible for complete beginners. I'm OK with using `blocking-subscribe` but not `QueueingConsumer` directly. The implementation of former can be updated if/when `QueueingConsumer` is removed from the Java client.